core/lib/zksync_core/src/api_server/web3/state.rs

```diff
@@ -638,6 +635,44 @@ impl Filters {
 
     /// Removes filter from the map.
     pub fn remove(&mut self, index: U256) -> bool {
-        self.state.remove(&index).is_some()
+        self.state.pop(&index).is_some()
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    #[test]
+    fn test_filters_functionality() {
+        use super::*;
+
+        let mut filters = Filters::new(2);
+
+        let filter1 = TypedFilter::Events(Default::default(), Default::default());
+        let filter2 = TypedFilter::Blocks(Default::default());
+        let filter3 = TypedFilter::PendingTransactions(Default::default());
+
+        let idx1 = filters.add(filter1.clone());
+        let idx2 = filters.add(filter2.clone());
+        let idx3 = filters.add(filter3.clone());
+
+        assert_eq!(filters.state.len(), 2);
+        assert_eq!(filters.state.contains(&idx1), false);
```

Comment:
Nit: `assert_eq` on Boolean values looks weird; it's more conventional to use `assert!(value)` / `assert!(!value)`.